src/paradigma/tremor/feature_extraction.py

```diff
@@ -0,0 +1,431 @@
+    
+    # construct mel-scale filters
+    freqs = np.linspace(melscale(fmin), melscale(fmax), num=n_filters+2) # equal intervals in mel scale
+    freqs_melscale = inverse_melscale(freqs) # convert to frequency domain
+    segment_length = segment_length_s * sampling_frequency 
+    filter_points = np.round((segment_length / sampling_frequency * freqs_melscale)).astype(int) + 1 # rounding the filter edges
+
+    filters = np.zeros((len(filter_points)-2, int(segment_length/2+1)))
+    for j in range(len(filter_points)-2):
+        filters[j, filter_points[j] : filter_points[j+2]] = signal.windows.triang(filter_points[j+2] - filter_points[j]) # triangular filters based on edges
+        filters[j,:] /= (sampling_frequency/segment_length * np.sum(filters[j,:])) # normalization of the filter coefficients
+    
+    # construct discrete cosine transform filters
+    dct_filters = np.empty((n_coefficients, n_filters))
+    dct_filters[0, :] = 1.0 / np.sqrt(n_filters)
+    samples = np.arange(1, 2 * n_filters, 2) * np.pi / (2.0 * n_filters)
+    for i in range(1, n_coefficients):
+        dct_filters[i, :] = np.cos(i * samples) * np.sqrt(2.0 / n_filters)
+
+    for spectrogram_window in spectrogram:
+        # mel-filtering
+        power_filtered = np.array([np.dot(filters, spectrogram_window[:, i]) for i in range(spectrogram_window.shape[1])])
+        
+        # taking the logarithm
+        log_power_filtered = np.log10(power_filtered)
+
+        # generate cepstral coefficients
+        cepstral_coefs = np.array([np.dot(dct_filters, x) for x in log_power_filtered])
+
+        # average coefficients over segments in window
+        cepstral_coefs_total.append(np.transpose(np.mean(cepstral_coefs,axis=0)))
+
+    return pd.DataFrame(cepstral_coefs_total, columns=['mfcc_{}'.format(j+1) for j in range(n_coefficients)])
+
+def extract_frequency_peak(
+    total_psd: pd.Series,
+    freq_vect: pd.Series,
+    fmin: float = 1,
+    fmax: float = 25
+    ) -> pd.Series:
+
+    """Extract the frequency of the peak in the power spectral density within the specified frequency band.
+    
+    Parameters
+    ----------
+    total_psd: pd.Series
+        The total power spectral density of the gyroscope signal
+    freq_vect: pd.Series
+        Frequency vector corresponding to the power spectral density
+    fmin: float
+        The lower bound of the frequency band in Hz (default: 1)
+    fmax: float
+        The upper bound of the frequency band in Hz (default: 25)
+        
+    Returns
+    -------
+    pd.Series
+        The frequency of the peak across windows
+    """
+    frequency_peak = []
+    
+    freq_range = freq_vect[0]
+    freq_idx = np.where((freq_range>=fmin) & (freq_range<=fmax))
+    for psd_window in total_psd: 
+        peak_idx = np.argmax(psd_window[freq_idx])
+        freq_peak = freq_range[peak_idx]+fmin
+        frequency_peak.append(freq_peak)
+
+    return frequency_peak
+
+def extract_low_freq_power(
```

Comment:
Deze functie loopt net als de vorige functie door de PSD. Je zou ze wellicht kunnen samenvoegen voor wat meer efficiëntie (of de loop uit de functie te halen).